editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -786,6 +787,44 @@ void SpriteFramesEditor::_animation_add() {
 	animations->grab_focus();
 }
 
+void SpriteFramesEditor::_animation_duplicate() {
+	if (!frames->has_animation(edited_anim)) {
+		return;
+	}
+
+	int counter = 1;
+	String name = String(edited_anim) + " " + itos(counter);
+	while (frames->has_animation(name)) {
+		counter++;
+		name = String(edited_anim) + " " + itos(counter);
+	}
+
+	undo_redo->create_action(TTR("Duplicate Animation"));
+	undo_redo->add_do_method(frames, "add_animation", name);
```

Comment:
It's not a big issue, but I wonder if it's the best to duplicate manually each property of the given resource. If we add a property on an animation, then this will break.
I think it would be better to modify the `SpriteFrames.animations` property directly instead, that should make it future-proof.

Reply:
can you explain deeper please, I think I didnt got it! what I got was: create a for loop and execute each action like the animation was being created manually?

Reply:
I mean, that's fine and it works. But if we someday add a new property to an animation, then it will break because you will be missing an action to copy the new property in the new create animation. So instead, I think you could use the `SpriteFrames.animations` property instead.
It would be something like:
```
undo_redo->add_undo_property(frames, "animations", frames.animations);
// Find the animation to duplicate in the frames.animations array, put it in the duplicated_animation variable.
...
//
Array animations = frames.animations;
animations.push_back(duplicated_animation); 
undo_redo->add_do_property(frames, "animations", animations);
```

But if you think it's too complex your solution works too, I don't mind much, it might just be a bit less future-proof.

Reply:
Anyway, I will approve the PR anyway, I think it's good to merge as-is.

Reply:
SpriteFrames have a struct named Anim, so it's easy to duplicate everything at once. But this needs to be done in SpriteFrames, something like `duplicate_animation(animation, new_animation)` and the code
```C++
void duplicate_animation(from, to) {
animation[to] = animations[from];
}
```
(structs are passed by value, so this should be enough)